src/transaction.rs

```diff
@@ -0,0 +1,220 @@
+use secp256k1::{PublicKey, Signature};
+use crate::{
+    slip::{Slip},
+    time::{create_timestamp}
+};
+
+/// A single record used in the history of transactions being routed around the network
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Hop {
+    pub from: PublicKey,
+    pub to: PublicKey,
+    sig: Signature,
+}
+
+impl Hop {
+    /// Creates a new `Hop`
+    ///
+    /// `to` - `secp256k1::PublicKey` address of where the transaction is headed
+    /// * `from` - `secp256k1::PublicKey` address of where the transaction came from
+    /// * `sign` - `secp256k1::Signature` verifying work done by routers
+    pub fn new(to: PublicKey, from: PublicKey, sig: Signature) -> Hop {
+        return Hop { to, from, sig }
+    }
+}
+/// A record containging data of funds between transfered between public addresses. It
+/// contains additional information as an optinal message field to transfer data around the network
+#[derive(Debug, PartialEq)]
+pub struct Transaction {
+    body: TransactionBody
+}
+
+/// Core data to be serialized/deserialized of `Transaction`
+#[derive(Debug, PartialEq)]
+pub struct TransactionBody {
+    id:       u64,
```

Comment:
id is not needed. The order of tx in the block encoding can represent the id.

Reply:
This ID isn't in reference to the id in the block but instead a globally unique transaction id. We want to record this as we can't discern it just from the order of just a single block

Reply:
Okay, but i we just use {blockid}-{txindex} then that can be used as a tx id which doesn't need to be stored in the tx, right?

Reply:
Good point. Will remove